packages/foam-vscode/src/core/janitor/generate-mdlinks.ts

```diff
@@ -0,0 +1,107 @@
+import { NoteLinkDefinition, Resource } from '../model/note';
+import { FoamWorkspace } from '../model/workspace';
+import { createMarkdownReferences } from '../services/markdown-provider';
+
+export interface TextReplace {
+  from: string;
+  to: string;
+}
+
+/**
+ * parse wikilink's rawText
+ * possible inpout:
+ *  1. [[\<filename\>#\<section\>|\<label\>]]
+ *  2. [[#\<section\>|\<label\>]]
+ *  3. [[\<filename\>#\<section\>]]
+ *  4. [[\<filename\>|\<label\>]]
+ */
+class WikilinkRawText {
+  private lable_m: string = '';
+  private filename_m: string = '';
+  private section_m: string = '';
+
+  public readFromText(definition: NoteLinkDefinition): WikilinkRawText {
+    const struct_rawText_regex =
+      /(?:(?<filename>[^\]|#]+)?(?:#(?<section>[^\]|]+))?)?(?:\|(?<label>[^\]]+))?/;
+    const match_result = definition.label.match(struct_rawText_regex);
+    if (match_result) {
+      this.filename_m = match_result[1];
+      this.section_m = match_result[2];
+      this.lable_m = match_result[3];
+    }
+    return this;
+  }
+
+  public getLable(): string {
+    if (this.lable_m) {
+      return this.lable_m;
+    }
+    // return `${this.getFilename()}${this.section_m ? '#' : ''}${this.getSection()}`;
+    return '';
+  }
+  public getFilename(): string {
+    if (this.filename_m) {
+      return this.filename_m;
+    }
+    return '';
+  }
+  public getSection(): string {
+    if (this.section_m) {
+      return this.section_m;
+    }
+    return '';
+  }
+  public toMarkdownLink(
```

Comment:
You probably can reuse the mechanisms available in `MarkdownLink` to achieve the same in a more integrated way